skribenten-backend/src/main/kotlin/no/nav/pensjon/brev/skribenten/services/BrevmalService.kt

```diff
@@ -0,0 +1,177 @@
+package no.nav.pensjon.brev.skribenten.services
+
+import io.ktor.server.application.*
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import no.nav.pensjon.brev.skribenten.services.BrevdataDto.BrevkategoriCode.*
+import no.nav.pensjon.brev.skribenten.services.BrevdataDto.BrevregeltypeCode
+import no.nav.pensjon.brev.skribenten.services.BrevdataDto.BrevregeltypeCode.*
+import no.nav.pensjon.brev.skribenten.services.PenService.SakType.*
+import org.slf4j.LoggerFactory
+
+class BrevmalService(
+    private val penService: PenService,
+    private val brevmetadataService: BrevmetadataService,
+) {
+    private val logger = LoggerFactory.getLogger(BrevmalService::class.java)
+
+    suspend fun hentBrevmalerForSak(
+        sakType: PenService.SakType,
+        includeEblanketter: Boolean,
+    ): List<LetterMetadata> = coroutineScope {
+        brevmetadataService.hentMaler(sakType, includeEblanketter).maler
+            .filter { filterForKontekst(it, isVedtaksKontekst = false) }
+            .filter { it.redigerbart }
+            .plus(brevmetadataService.hentMaler(sakType, includeEblanketter).eblanketter)
```

Comment:
Har BrevmetadataService caching? Om ikke så henter du vel samme ting to ganger her?

Reply:
Ah. Var nokk en inline som gikk skeis. Ifix